fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -233,17 +253,21 @@ def predict(self, X):
             the result will be a scalar. Otherwise the result will be an
         :rtype: Scalar or array
         """
+        if self.best_result is None:
+            raise NotFittedException(_NO_PREDICT_BEFORE_FIT)
         return self.best_result.predictor.predict(X)
 
     def predict_proba(self, X):
-        """Provides the result of predict_proba from the
+        """Provides the result of ``predict_proba`` from the
         best model found by the grid search. The underlying
-        estimator must support predict_proba for this
+        estimator must support ``predict_proba`` for this
         to work.
 
         :param X: The data for which predictions are required
         :type X: Array
         """
+        if self.best_result is None:
+            raise NotFittedException(_NO_PREDICT_BEFORE_FIT)
```

Comment:
should this be self._best_result?

Reply:
It could be - this will read it via the property